### PR TITLE
rest: add initial support for PATCH

### PR DESCRIPTION
**Pull request by filariow**
Signed-off-by: Francesco Ilario <filario@redhat.com>

### CODE REVIEW

File: server/core/workspace/workspace_patch.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	// reply
+	return &PatchWorkspaceResponse{
+		Workspace: pw,
+	}, nil
+}
+
+func (h *PatchWorkspaceHandler) applyPatch(w *workspacesv1alpha1.Workspace, command PatchWorkspaceCommand) (*workspacesv1alpha1.Workspace, error) {
+	if command.PatchType != types.MergePatchType {
+		return nil, fmt.Errorf("unsupported patch type: %s", command.PatchType)
+	}
```
Comment (sadlerap) [inline server/core/workspace/workspace_patch.go:85]: maybe add a test so that other patch types are rejected for now?

Comment (filariow) [reply server/core/workspace/workspace_patch.go:85]: good idea, let me add them

Comment (filariow) [reply server/core/workspace/workspace_patch.go:85]: done in https://github.com/konflux-workspaces/workspaces/pull/315/commits/bd8a3d359adbeede0a66dd20ad799c922d1a8c04

**Review by sadlerap — APPROVED** [review]
I think this is okay.  I have some concerns about the soundness of the semantics, but I think this should be fine for a demo.

**Pull request merged**